eth2util/keymanager/keymanager.go

```diff
@@ -0,0 +1,131 @@
+// Copyright © 2022 Obol Labs Inc.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of  MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package keymanager provides ETH2 keymanager API (https://ethereum.github.io/keymanager-APIs/) functionalities.
+package keymanager
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/z"
+	"github.com/obolnetwork/charon/eth2util/keystore"
+)
+
+// New returns a new Client.
+func New(url string) Client {
+	return Client{
+		baseURL: url,
+	}
+}
+
+// Client is the REST client for keymanager API requests.
+type Client struct {
```

Comment:
Why are we having a struct with only one field? why can we not simple pass baseURL as a function parameter and have pure functions?

Reply:
We want to introduce a REST API client for [keymanager APIs](https://ethereum.github.io/keymanager-APIs/) just like we have go-eth2-client for ethereum beacon APIs.